In [19]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
pd.set_option('display.max_columns', None)

In [9]:
data = pd.read_excel('music data.xlsx')

In [20]:
data.head()

ID            Full_song_name       Author              Title  danceability  \
0   0  10Cc - Dreadlock Holiday         10Cc  Dreadlock Holiday         0.837   
1   1         2-4 Family - Stay   2-4 Family               Stay         0.858   
2   2   2 Eivissa - Oh La La La    2 Eivissa        Oh La La La         0.789   
3   3    2 Unlimited - No Limit  2 Unlimited           No Limit         0.650   
4   4      2 Unlimited - No One  2 Unlimited             No One         0.828   

   energy key  key_code  mode  loudness  speechiness  acousticness  \
0   0.380   G         7     0   -13.341       0.0640       0.54100   
1   0.715  F#         6     1    -5.764       0.0827       0.07010   
2   0.862   F         5     0    -4.060       0.0398       0.02630   
3   0.985   F         5     0   -14.404       0.0445       0.00872   
4   0.714   B        10     0   -10.100       0.0457       0.02720   

   instrumentalness  liveness  valence    tempo  duration_ms  time_signature  \
0          0.007890    0.1980    0.892  104.995       267947               4   
1          0.000000    0.8350    0.853   98.023       231000               4   
2          0.036800    0.2190    0.882  130.442       211253               4   
3          0.004220    0.0485    0.757  141.047       224221               4   
4          0.000494    0.0611    0.969  132.931       205910               4   

   Antyradio  Chillzet  Czwórka  Dwójka  Eska  Jedynka  RMF_Classic  RMF_FM  \
0         10         0        0       0     0        6            0       1   
1          0         0        0       0     0        0            0       0   
2          0         0        3       0     0        0            0       9   
3          0         0        2       0     0        0            0       6   
4          0         0        0       0     0        0            0       0   

   RMF_MAXXX  Trójka  VOX_FM  ZET  Złote_Przeboje  Total_plays  Release_year  \
0          0      23     190    0             775         1005          1978   
1          0       0     287    0               0          287          1999   
2          0       0     233    0               0          245          2006   
3          1       0     485    2               0          496          1993   
4          0       0     346    0               0          346          1994   

   Antyradio_year_mean  Chillzet_year_mean  Czwórka_year_mean  \
0                    1                   0                  0   
1                    0                   0                  0   
2                    0                   0                  0   
3                    0                   0                  0   
4                    0                   0                  0   

   Dwójka_year_mean  Eska_year_mean  Jedynka_year_mean  RMF_Classic_year_mean  \
0                 0               0                  0                      0   
1                 0               0                  0                      0   
2                 0               0                  0                      0   
3                 0               0                  0                      0   
4                 0               0                  0                      0   

   RMF_FM_year_mean  RMF_MAXXX_year_mean  Trójka_year_mean  VOX_FM_year_mean  \
0                 0                    0                 2                19   
1                 0                    0                 0                28   
2                 0                    0                 0                23   
3                 0                    0                 0                48   
4                 0                    0                 0                34   

   ZET_year_mean  Złote_Przeboje_year_mean  Suma_odtworzen_year_mean  \
0              0                        77                       100   
1              0                         0                        28   
2              0                         0                        24   
3          

In [11]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2477 entries, 0 to 2476
Data columns (total 49 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   ID                        2477 non-null   int64  
 1   Full_song_name            2477 non-null   object 
 2   Author                    2477 non-null   object 
 3   Title                     2477 non-null   object 
 4   danceability              2477 non-null   float64
 5   energy                    2477 non-null   float64
 6   key                       2477 non-null   object 
 7   key_code                  2477 non-null   int64  
 8   mode                      2477 non-null   int64  
 9   loudness                  2477 non-null   float64
 10  speechiness               2477 non-null   float64
 11  acousticness              2477 non-null   float64
 12  instrumentalness          2477 non-null   float64
 13  liveness                  2477 non-null   float64
 14  valence 

In [23]:
data.duplicated().sum()

0

In [24]:
data.isnull().sum()

ID                          0
Full_song_name              0
Author                      0
Title                       0
danceability                0
energy                      0
key                         0
key_code                    0
mode                        0
loudness                    0
speechiness                 0
acousticness                0
instrumentalness            0
liveness                    0
valence                     0
tempo                       0
duration_ms                 0
time_signature              0
Antyradio                   0
Chillzet                    0
Czwórka                     0
Dwójka                      0
Eska                        0
Jedynka                     0
RMF_Classic                 0
RMF_FM                      0
RMF_MAXXX                   0
Trójka                      0
VOX_FM                      0
ZET                         0
Złote_Przeboje              0
Total_plays                 0
Release_year                0
Antyradio_

In [ ]:

# wykres z unique counts sns.countplot(data[kolumna/y]).unique()
# Filtrowanie zachowując jakąś logikę data[data.danceability >0.5].head() - zrobić takich kilka. Można zrobić na grafie porównanie ile unikatowych tytułów ma każde radio zagranych.
# Przy okazji zrobić dla kilu kolumn ze spotify jakiąś statystykę opisową. Mediana, decyle, kwartyle itd.
# I oczywiście korelacje z wykorzystaniem SNSa sns.heatmap(data.corr())
    

In [74]:
print(len(data.Author.unique()))
print(2477/len(data.Author.unique()))

1275
1.9427450980392156


In [90]:
sum(i > 1 for i in data.Author.value_counts())

444

#### The 2 500 songs played between 2012 and 2022 belong to 1 275 artists. That means that on average each artist had almost 2 songs played (possibly miltiple times) during that period. Realistically there was only 444 artists whose more than one song was played, which means that over 800 authors had only one song played.

In [91]:
radio_list = ['Antyradio','Chillzet','Czwórka','Dwójka','Eska','Jedynka','RMF_Classic','RMF_FM','RMF_MAXXX','Trójka','VOX_FM','ZET','Złote_Przeboje']

In [102]:
def unique_authors_per_radio(radio_name):
    return len(data[data[radio_name] > 0].Author.unique())
    

In [141]:
unique_authors = pd.DataFrame(columns = ['Radio', 'Unique_authors'])
for i in radio_list:
    new_row = pd.Series({'Radio': i, 'Unique_authors': unique_authors_per_radio(i)})
    unique_authors = pd.concat([unique_authors, new_row.to_frame().T], ignore_index = True)
    print(i, ": ", unique_authors_per_radio(i))

Antyradio :  334
Chillzet :  591
Czwórka :  487
Dwójka :  45
Eska :  486
Jedynka :  669
RMF_Classic :  138
RMF_FM :  682
RMF_MAXXX :  419
Trójka :  637
VOX_FM :  509
ZET :  685
Złote_Przeboje :  487


In [142]:
unique_authors

Radio Unique_authors
0        Antyradio            334
1         Chillzet            591
2          Czwórka            487
3           Dwójka             45
4             Eska            486
5          Jedynka            669
6      RMF_Classic            138
7           RMF_FM            682
8        RMF_MAXXX            419
9           Trójka            637
10          VOX_FM            509
11             ZET            685
12  Złote_Przeboje            487